### DISCUSSION TITLE
Is it possible to edit block without pushing edit to undo stack

### DISCUSSION THREAD

**Discussion by mukeshgurpude** [WordPress/gutenberg#42835] [Q&A]
When you use [updateBlockAttributes](https://developer.wordpress.org/block-editor/reference-guides/data/data-core-block-editor/#updateblockattributes) to update the block content, it pushes the edit to the editor's History Stack.

Is any of the following options feasible?
- Content editing without committing to history
- removing an edit entry from the Undo stack

**Comment by t-hamano** [comment]
It is not a stable API, but what about using `__unstableMarkNextChangeAsNotPersistent`?
It is also used in some core blocks to prevent the creation of Undo levels.
